src/SprykerSdk/SyncApi/Console/OpenApiSchemaMergerConsole.php

```diff
@@ -0,0 +1,101 @@
+use Symfony\Component\Yaml\Yaml;
+
+class OpenApiSchemaMergerConsole extends AbstractConsole
+{
+    /**
+     * @var string
+     */
+    public const COMMAND_NAME = 'openapi:schema:merge';
+
+    /**
+     * @var string
+     */
+    public const DESCRIPTION = 'This command merges OpenApi Specs from several sources and a root definition into one.';
+
+    protected function configure(): void
+    {
+        $this->setName(static::COMMAND_NAME);
+        $this->setDescription(static::DESCRIPTION);
+
+        $this->addArgument('source', InputArgument::REQUIRED, 'The root file of the OpenApi schema which will be used to merge with other schemas.');
+        $this->addArgument('target', InputArgument::REQUIRED, 'The target file name that should be created after merge.');
+
+        $this->addOption('additional-schemas', 'a', InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY, 'The additional OpenApi schema files that should be merged with the root schema.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $source = sprintf('%s/%s', getcwd(), $input->getArgument('source'));
+        $target = sprintf('%s/%s', getcwd(), $input->getArgument('target'));
+
+        $additionalSchemas = $input->getOption('additional-schemas');
+
+        $yaml = new Yaml();
+        $sourceSchema = $yaml->parseFile($source);
+
+        $paths = $sourceSchema['paths'] ?? [];
+        $schemas = $sourceSchema['components']['schemas'] ?? [];
+        $parameters = $sourceSchema['components']['parameters'] ?? [];
+
+        foreach ($additionalSchemas as $additionalSchema) {
+            $additionalSchema = sprintf('%s/%s', getcwd(), $additionalSchema);
+            $coreSchema = $yaml->parseFile($additionalSchema);
+            $paths = $this->recursiveMerge($coreSchema['paths'] ?? [], $paths);
+            $schemas = $this->recursiveMerge($coreSchema['components']['schemas'] ?? [], $schemas);
+            $parameters = $this->recursiveMerge($coreSchema['components']['parameters'] ?? [], $parameters);
+        }
+
+        $mergedOpenApi = new OpenApi([
+            'openapi' => $sourceSchema['openapi'],
+            'info' => $sourceSchema['info'],
+            'servers' => $sourceSchema['servers'],
+            'paths' => $paths,
+            'components' => [
+                'schemas' => $schemas,
+                'parameters' => $parameters,
+            ],
+            'security' => $sourceSchema['security'] ?? [],
+            'tags' => $sourceSchema['tags'] ?? [],
+            'externalDocs' => $sourceSchema['externalDocs'] ?? [],
+        ]);
+
+        // Save the merged specification to a file
+        Writer::writeToYamlFile($mergedOpenApi, $target);
+
+        return static::CODE_SUCCESS;
+    }
+
+    /**
+     * @param array $array1
+     * @param array $array2
+     *
+     * @return array
+     */
+    protected function recursiveMerge(array $array1, array $array2): array
```

Comment:
How is it different from `array_merge_recursive` ?

Reply:
The `recursiveMerge` function overwrites the values in the first array with the values from the second array when keys match.

`array_merge_recursive` combines values into arrays when keys match, rather than overwriting them.